frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,259 @@ pub struct PoolRoles<AccountId> {
+				|p| {
+					// `min_delay` blocks must have been surpassed since last update.
+					if <frame_system::Pallet<T>>::block_number().saturating_sub(p) <
+						t.change_rate.min_delay
+					{
+						return true
+					}
+					// ensure the `max_increase` durations surpassed since the previous
+					// commission update allow the attempted commission increase.
+					//
+					// the attempted increase in commission relative to the current commission.
+					let attempted_increase = (*to).saturating_sub(self.commission_or_zero());
+
+					// the total durations passed since the last commission update.
+					let intervals_passed = self.intervals_since_block(&p, &t.change_rate.min_delay);
+
+					// the maximum allowed increase, where the current `max_increase`
+					// Perbill is converted into a u32 by multiplying itself with 100_u32
+					// and multiplied by durations passed, before being converted back
+					// into a Perbill.
+					let max_allowed_increase = Perbill::from_percent(
+						(t.change_rate.max_increase * 100_u32) * intervals_passed,
+					);
+
+					// throttled (true) if attempted increase is greater than the maximum
+					// allowed increase, as a percentage.
+					attempted_increase > max_allowed_increase
+				},
+			) {
+				return true
+			}
+		}
+		false
+	}
+
+	/// Set the pool's commission.
+	///
+	/// Update commission based on `commission` and `payee`. Do not allow a
+	/// commission above global maximum if set. If throttle is present, record the
+	///  current block as the previously updated commission. If the supplied
+	/// commission is zero, `None` will be inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		commission: &Perbill,
+		payee: T::AccountId,
+	) -> DispatchResult {
+		ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+		ensure!(
+			GlobalMaxCommission::<T>::get().map_or(true, |m| commission <= &m),
+			Error::<T>::GlobalMaxCommissionExceeded
+		);
+		ensure!(self.max.map_or(true, |m| commission <= &m), Error::<T>::CommissionExceedsMaximum);
+
+		self.current = Some((*commission, payee));
+
+		// update `throttle.previous_set_at` if a throttle has been configured.
+		let _ = self
+			.throttle
+			.as_mut()
+			.map(|t| t.register_change(<frame_system::Pallet<T>>::block_number()));
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	///
+	/// The pool's maximum commission can initially be set to any value, below
+	/// global maximum if set, and only smaller values thereafter. If larger
+	/// values are attempted, this function will return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then
+	/// `current.0` will also be updated to the new maximum.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		ensure!(
+			GlobalMaxCommission::<T>::get().map_or(true, |m| new_max <= m),
+			Error::<T>::GlobalMaxCommissionExceeded
+		);
```

Comment:
```suggestion
```

I am not so sure about this. `GlobalMaxCommission` is an upper bound for all the current commissions. I don't think it needs to influence this maximum. 

e.g. `GlobalMaxCommission = 2%` and a pool having its local max ever to 10% is a valid configuration.

Reply:
OK, a bit of extra work on the UI to ensure the correct max is displayed, but removing will save a storage read.